test/widgets/inbox_test.dart

```diff
@@ -385,6 +450,50 @@ void main() {
           checkAppearsUncollapsed(tester, 1, findSectionContent);
         });
 
+        testWidgets('collapse stream section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final topicFinder = find.text('stream 1 topic 4' ).hitTestable();
+          final listFinder = find.byType(Scrollable);
+
+          // Scroll part of [_StreamSection] offscreen.
+          await dragUntilInvisible(
+            tester, topicFinder, listFinder, const Offset(0, -50));
+
+          final headerRow = findStreamHeaderRow(tester, 1);
+          // Check that the header is present (which must therefore
+          // be as a sticky header).
+          check(headerRow).isNotNull();
+
+          await tapCollapseIcon(tester, 1);
+
+          final headerRowAfterTap = findStreamHeaderRow(tester, 1);
+          // Check that the header is still visible even after
+          // collapsing the section.
+          check(headerRowAfterTap).isNotNull();
+        });
+
+        testWidgets('collapse stream section in the middle of screen', (tester) async {
+          await setupVarious(tester);
+
+          final headerRow = findStreamHeaderRow(tester, 1);
+          final rectBeforeTap = tester.getRect(find.byWidget(headerRow!));
+
+          // Check that the header is present (which must therefore
+          // be somewhere in the middle of screen)
+          check(headerRow).isNotNull();
```

Comment:
The `isNotNull` check should come before the `!` — otherwise it's impossible for the check to fail, because if `headerRow` were null we'd have already thrown at `headerRow!`.